libs/application/templates/driving-instructor-registrations/src/fields/StudentsOverview/index.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState, useEffect } from 'react'
+import {
+  Button,
+  Box,
+  Input,
+  Pagination,
+  Stack,
+} from '@island.is/island-ui/core'
+import { Table as T } from '@island.is/island-ui/core'
+import { PAGE_SIZE, pages, paginate } from './pagination'
+import ViewStudent from '../ViewStudent/index'
+import { Application } from '@island.is/application/core'
+import { m } from '../../lib/messages'
+import { useLocale } from '@island.is/localization'
+import FindStudentModal from '../FindStudentModal/index'
+import { useQuery } from '@apollo/client'
+import { InstructorsStudentsQuery } from '../../graphql/queries'
+import Skeleton from './Skeleton'
+import * as styles from '../style.css'
+
+interface Data {
+  application: Application
+}
+
+interface Student {
+  id: string
+  name: string
+  nationalId: string
+  totalLessonCount: number
+}
+
+const StudentsOverview = ({ application }: Data) => {
+  const { formatMessage } = useLocale()
+
+  /* table pagination */
+  const [page, setPage] = useState(1)
+  const [totalPages, setTotalPages] = useState(0)
+  const { data, loading } = useQuery(InstructorsStudentsQuery)
+
+  const [pageStudents, setPageStudents] = useState(
+    data ? (data.drivingLicenseBookStudentsForTeacher as Array<Student>) : [],
+  )
+
+  /* table view */
+  const [showTable, setShowTable] = useState(true)
+  const [studentId, setStudentId] = useState('')
+  const [searchTerm, setSearchTerm] = useState('')
+  const [filteredStudents, setFilteredStudents] = useState<Array<object>>()
+
+  const handlePagination = (page: number, students: Array<object>) => {
+    setPage(page)
+    setTotalPages(pages(students?.length))
+    setFilteredStudents(students)
+    setPageStudents(paginate(students, PAGE_SIZE, page))
+  }
+
+  const filter = (searchTerm: string) => {
+    if (searchTerm.length) {
+      const filteredList = data.drivingLicenseBookStudentsForTeacher?.filter(
+        (student: Student) =>
+          student.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
+          student.nationalId.includes(searchTerm),
+      )
+
+      handlePagination(1, filteredList)
+    } else {
+      handlePagination(1, data?.drivingLicenseBookStudentsForTeacher)
+    }
+  }
+
+  useEffect(() => {
+    filter(searchTerm)
```

Comment:
Shouldn't there be a delay on this?
Otherwise there will be x queries fired for a string of length x

Reply:
If we add delay, would the user experience be worse?

Reply:
Nvm, changed to debounce with 500 ms delay. How do you like https://github.com/island-is/island.is/pull/6481/commits/38464708d822da50c21bb53070780a8005d9f6f3?